design/apidsl/media_type.go

```diff
@@ -355,6 +364,8 @@ func Link(name string, view ...string) {
 	}
 }
 
+// TODO: is this correct?
+// CollectionOf can be used in: Media, Response, ResponseTemplate
```

Comment:
There are more places, maybe it's simpler to say something like `wherever a media type can`, e.g `Attribute("foo", CollectionOf(Bar)`